### DISCUSSION TITLE
[Feature request] Pointer emulation

### DISCUSSION THREAD

**Discussion by itshog** [YaLTeR/niri#726] [Ideas]
Would it be possible to add touch pointer emulation to Niri? By that I mean the ability to simulate mouse clicks with touch/pen input in clients which don't support touch input directly (for instance [yambar](https://codeberg.org/dnkl/yambar), see [this issue](https://codeberg.org/dnkl/yambar/issues/359))

If I'm not mistaken [this](https://github.com/swaywm/sway/pull/5229) is the implementation of pointer emulation in Sway. I didn't find anything in Smithay or Cosmic upstream though.

I'd be happy to help in testing this functionality.

**Comment by YaLTeR** [comment]
I guess, though I'm not sure if something might be needed on smithay side too.

**Reply by itshog** [reply]
Any updates on this? I mistakenly duplicated my own feature request: I recently switched to [i3bar-river](https://github.com/MaxVerevkin/i3bar-river) since [yambar](https://codeberg.org/dnkl/yambar) isn't maintained anymore and I'm facing the same problem.

See for reference [this issue](https://github.com/MaxVerevkin/i3bar-river/issues/23) I created on the i3bar-river repo and the answer of the author. [This](https://codeberg.org/river/river/issues/1025) is instead the relevant issue on River side, which also lack pointer emulation at the moment.

**Reply by YaLTeR** [reply]
No updates. I don't really use touch so uh

**Reply by itshog** [reply]
Would it be possible to implement this in a similar way to what Sway does? In my opinion this feature and touchscreen gestures (discussion #463) would make Niri almost the de-facto standard compositor for foldable laptops (at least in the realm of standalone compositors), as its scrolling paradigm works charm both with touchpad/keyboard and touch input. At the moment though the only way to control windows placement and sizing and in general to move across workspaces when using a laptop in tablet mode is to have a big, space-consuming bar (with a toolkit that directly supports touch events, so basically only [waybar](https://github.com/Alexays/Waybar) comes to mind) with ugly buttons to scroll left/right etc.

At least having pointer emulation would permit the user to choose from a wider range of bars and layer-shell components, and also make sure that every legacy app which maybe doesn't support touch events (or any app using alternative toolkits) can be used with a folded laptop.

I would be eager to help in testing and debugging those two features. I also understand that maybe the project priorities right now aren't focused on touch/tablet support, which is probably still a somewhat niche use-case, but I believe Niri's potential on tablet/laptops hybrid shouldn't be underestimated.

**Reply by YaLTeR** [reply]
I'm doing this on my free time, so my priorities are what I personally use and am interested in. "Potential" on other devices doesn't magically add time or motivation (and also unfortunately doesn't translate into manufacturers lining up to pay for development work).

If someone makes a PR, I can review it, and testing and debugging will be appreciated.

**Reply by itshog** [reply]
I didn't mean to sound entitled, the work of the developers on this project is already amazing as it stands. I'd really like to contribute in making a PR for those features, but as of now my programming knowledge is basically non-existent, more so when it comes to Rust. If something needs testing on touch devices I'll gladly help though.
